### ISSUE TITLE
desisim/spec_qa/s2n: don't read files multiple times

### ISSUE THREAD

**Issue by sbailey** [desihub/desisim#444]
While debugging a separate issue, I found that `desi_qa_s2n` is quite slow because it keeps reading in the same truth and cframe files over and over again via `load_s2n_values(objtype, nights, channel)`.

Refactor to read in the data only once, and then filter as needed for various objtype and channels.
